### PR TITLE
VB translation of Refactoring  - allow members ordering

### PR DESCRIPTION
**Pull request by jwooley**
Translating the Allow Members Ordering refactoring analysis and code Fix.

Also, removing the CSharp.Workspaces library from the VB test project.

### CODE REVIEW

File: build.targets.ps1 [hunk 1]
```diff
@@ -195,4 +195,5 @@ function RunTestWithCoverage($fullTestDllPaths) {
         Write-Host -ForegroundColor DarkBlue "Uploading coverage report to Coveralls.io"
         Exec { . $converallsNetExe --opencover $outputXml }
     }
+}
```
Comment (giggio) [inline build.targets.ps1]: This change is breaking the build.

**Comment by giggio** [comment]
Build is failing. https://ci.appveyor.com/project/code-cracker/code-cracker/build/1.0.0.345

File: src/CSharp/CodeCracker/Usage/RethrowExceptionAnalyzer.cs [hunk 2]
```diff
@@ -24,7 +24,7 @@ public class RethrowExceptionAnalyzer : DiagnosticAnalyzer
             MessageFormat,
             Category,
             DiagnosticSeverity.Warning,
-            isEnabledByDefault: true,
+            isEnabledByDefault: false,
```
Comment (giggio) [inline src/CSharp/CodeCracker/Usage/RethrowExceptionAnalyzer.cs]: why was this changed?

File: test/VisualBasic/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Base.vb [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+'        Public Delegate Function doubleDelegate(num As Double) As Double
+'        Public Sub New()
+'            Prop = ""
+'            Field = ""
+'        End Sub
+
+'        Public Event EventField1 As Action
+'        Public field As String
+'    End Class
+'End Namespace
```
Comment (giggio) [inline test/VisualBasic/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Base.vb]: Remove commented code.

File: test/VisualBasic/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Stylecop.vb [hunk 4]
```diff
@@ -0,0 +1,563 @@
...
+'            Field = ""
+'        End Sub
+'        Public Event EventField1 As Action
+'        Public Event EventField As Action
+'        Public Field As String
+'        Public Shared Field1 As String
+'    End Class
+
+'End Namespace
+
```
Comment (giggio) [inline test/VisualBasic/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Stylecop.vb]: More commented code to remove.

File: build.targets.ps1 (hunk 1 above)
Comment (jwooley) [inline build.targets.ps1]: Not an intentional change. Must have been a merge issue. This change can be reverted.

File: src/CSharp/CodeCracker/Usage/RethrowExceptionAnalyzer.cs (hunk 2 above)
Comment (jwooley) [inline src/CSharp/CodeCracker/Usage/RethrowExceptionAnalyzer.cs]: Merge issue. I didn't change it manually.

File: test/VisualBasic/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Base.vb (hunk 3 above)
Comment (jwooley) [inline test/VisualBasic/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Base.vb]: Will do.

**Comment by jwooley** [comment]
Applying requested updates for previous pull request

File: test/VisualBasic/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Base.vb [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+Public {0} Foo
+    {2}
+    {1}
+End {0}", typeDeclaration, memberA, memberB)
+
+            Await VerifyBasicFixAsync(source, expected, codeFixProvider:=codeFixProvider)
+            Assert.True(codeFixProvider.HasIComparerBeenCalled, "The IComparer must be used to sort the members of that type")
+        End Function
+
+        <Fact>
```
Comment (giggio) [inline test/VisualBasic/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.Base.vb]: This should be a `<Theory>`.

**Comment by giggio** [comment]
A test is failing: https://ci.appveyor.com/project/code-cracker/code-cracker/build/1.0.0.353/tests

````
System.InvalidOperationException : [Fact] methods are not allowed to have parameters. Did you mean to use [Theory]?
````

**Comment by giggio** [comment]
I can see that you merged this into #247. I will close this one, then.
